game-core/src/main/java/org/triplea/game/server/HeadlessServerSetupPanelModel.java

```diff
@@ -21,7 +39,64 @@
 
   @Override
   public void showSelectType() {
-    new ServerModel(gameSelectorModel, this, null).createServerMessenger();
+    new ServerModel(gameSelectorModel, this, null, new LaunchAction() {
+      @Override
+      public void handleGameInterruption(final GameSelectorModel gameSelectorModel, final ServerModel serverModel) {
+        try {
+          log.info("Game ended, going back to waiting.");
+          // if we do not do this, we can get into an infinite loop of launching a game,
+          // then crashing out, then launching, etc.
+          serverModel.setAllPlayersToNullNodes();
+          final File f1 = getAutoSaveFileUtils().getHeadlessAutoSaveFile();
+          if (!f1.exists() || !gameSelectorModel.load(f1)) {
+            gameSelectorModel.resetGameDataToNull();
+          }
+        } catch (final Exception e1) {
+          log.log(Level.SEVERE, "Failed to load game", e1);
+          gameSelectorModel.resetGameDataToNull();
+        }
+      }
+
+      @Override
+      public void onGameInterrupt() {
+        // tell headless server to wait for new connections:
```

Comment:
Comment is a bit of a code smell as it describes a single line that should perhaps describe itself. Can we reasonably rename the method `waitForNewConnections()` and remove the comment? If not, would the comment be better off as a TODO comment to rename the method?